proposed/building-arch-specific-applications.md

```diff
@@ -0,0 +1,806 @@
+
+* Architecture-agnostic, framework-dependent, application that includes native dependencies for all supported architectures and no EXE
+* Achitecture-specific, framework-dependent, application that includes an EXE and native dependencies for a single targeted architecture
+* Architecture-specific, self-contained, application that includes EXE and native dependencies for a single targeted architecture
+
+The proposal is:
+
+* Retain the first behavior as the default.
+* Enable the second behavior as opt-in. Enable this behavior in Windows client application templates.
+* Maintain the third behavior as opt-in (change defaults for `-r` and `--self-contained` for 3.0+).
+
+The first behavior is most similar to the .NET Core 2.2 behavior. It is a good behavior because it builds an application with no assumptions and produces artifacts with the broadest scope and utility. Many developers will want the second behavior. It needs to be very easy to opt-in to that behavior, either via the CLI or with a single property in a template.
+
+## Detailed Proposal
+
+Many .NET Framework developers unfamiliar with .NET Core will start using the newer environment with .NET Core 3.0 and will expect a similar experience as they have been used to, particularly for those that only ever intend to target Windows with Windows Forms and WPF applications. Very few of them will expect Linux or macOS binaries in their build output. We need to deliver an experience they expect.
+
+### Goals
+
+The following broad goals should be satisfied for building applications with native dependencies:
+
+* .NET CLI defaults (for architecture and native dependencies) are intuitive and align with key user scenarios.
+* .NET CLI opt-in arguments (for architecture and native dependencies) have good usability for key user scenarios, with a bias to short and memorable arguments and good defaults for those arguments.
+* It is straightforward to transform CLI opt-in arugments (for architecture and native dependencies) that enable your project use-case into msbuild properties in your project file, resulting in a custom default for that project.
+* It is straightforward to produce deterministic builds, independent of OS or architecture of the build machine.
+* It is easy and intutive to share applications with others or deploy them to other machines (like a Rasperry Pi, or a Windows ARM64 laptop) as part of a fluid workflow.
+
+The following are more specific goals to improve CLI usability:
+
+* `build` and `publish` should have the same capability, except where there is a good scenario reason why a task should only be made available via `publish`.
+* Defaults should be the same between `build` and `publish`.
+* All common scenarios should work without specifying a RID if you are happy targeting the current machine.
+* We should strongly avoid any breaking changes.
+
+### Proposed Application Types
+
+For the purposes of this document, there are three types of applications:
```

Comment:
This feels repetitive from above and you are already adding details below. Maybe a table or sub bullets would make it easier to see how these three align

Reply:
I did repeat myself. I intentionally created a summary at the start and then a more indepth-version.  I streamlined the earlier version a bit to help.